WordPress/gutenberg
# Components: Refactor `MenuGroup` tests to `@testing-library/react`

## What?
We've recently started refactoring `enzyme` tests to `@testing-library/react`.

This PR refactors the `<MenuGroup />` component tests from `enzyme` to `@testing-library/react`.

## Why?
`@testing-library/react` provides a better way to write tests for accessible components that is closer to the way the user experiences them.

## How?
We're straightforwardly replacing `enzyme` tests with `@testing-library/react` ones, using `jest-dom` matchers and screen queries to find elements.

We're also removing an unnecessary snapshot test and replacing it with user-oriented functionality tests.

## Testing Instructions
Verify tests pass: `npm run test:unit packages/components/src/menu-group/test/index.js`
